access/email/email_test.go

```diff
@@ -0,0 +1,653 @@
+		t.Skip("Doesn't work in OSS version")
+	}
+
+	s.startApp()
+
+	req := s.createAccessRequest([]string{s.userNames.reviewer1})
+	s.checkPluginData(req.GetName(), func(data PluginData) bool {
+		return len(data.EmailThreads) > 0
+	})
+
+	s.skipMessages(s.Context(), t, 2)
+
+	s.reviewer1().SubmitAccessRequestReview(s.Context(), req.GetName(), types.AccessReview{
+		Author:        s.userNames.reviewer1,
+		ProposedState: types.RequestState_APPROVED,
+		Created:       time.Now(),
+		Reason:        "okay",
+	})
+
+	messages := s.getMessages(s.Context(), t, 2)
+
+	reply := messages[0].Body
+
+	assert.Contains(t, reply, s.userNames.reviewer1+" reviewed the request", "reply must contain a review author")
+	assert.Contains(t, reply, "Resolution: ✅ APPROVED", "reply must contain a proposed state")
+	assert.Contains(t, reply, "Reason: okay", "reply must contain a reason")
+
+	s.reviewer2().SubmitAccessRequestReview(s.Context(), req.GetName(), types.AccessReview{
+		Author:        s.userNames.reviewer2,
+		ProposedState: types.RequestState_DENIED,
+		Created:       time.Now(),
+		Reason:        "not okay",
+	})
+
+	messages = s.getMessages(s.Context(), t, 2)
+
+	reply = messages[0].Body
+
+	assert.Contains(t, reply, s.userNames.reviewer2+" reviewed the request", "reply must contain a review author")
+	assert.Contains(t, reply, "Resolution: ❌ DENIED", "reply must contain a proposed state")
+	assert.Contains(t, reply, "Reason: not okay", "reply must contain a reason")
+}
+
+func (s *EmailSuite) TestApprovalByReview() {
+	t := s.T()
+
+	if !s.teleportFeatures.AdvancedAccessWorkflows {
+		t.Skip("Doesn't work in OSS version")
+	}
+
+	s.startApp()
+
+	req := s.createAccessRequest([]string{s.userNames.reviewer2})
+
+	s.skipMessages(s.Context(), t, 2)
+
+	s.reviewer1().SubmitAccessRequestReview(s.Context(), req.GetName(), types.AccessReview{
+		Author:        s.userNames.reviewer1,
+		ProposedState: types.RequestState_APPROVED,
+		Created:       time.Now(),
+		Reason:        "okay",
+	})
+
+	messages := s.getMessages(s.Context(), t, 2)
+
+	assert.Contains(t, messages[0].Body, s.userNames.reviewer1+" reviewed the request", "reply must contain a review author")
+
+	s.reviewer2().SubmitAccessRequestReview(s.Context(), req.GetName(), types.AccessReview{
+		Author:        s.userNames.reviewer2,
+		ProposedState: types.RequestState_APPROVED,
+		Created:       time.Now(),
+		Reason:        "finally okay",
+	})
+
+	messages = s.getMessages(s.Context(), t, 2)
+	assert.Contains(t, messages[0].Body, s.userNames.reviewer2+" reviewed the request", "reply must contain a review author")
+
+	messages = s.getMessages(s.Context(), t, 2)
+	assert.Contains(t, messages[0].Body, "Status: ✅ APPROVED (finally okay)")
+}
+
+func (s *EmailSuite) TestDenialByReview() {
+	t := s.T()
+
+	if !s.teleportFeatures.AdvancedAccessWorkflows {
+		t.Skip("Doesn't work in OSS version")
+	}
+
+	req := s.createAccessRequest([]string{s.userNames.requestor})
+
+	s.startApp()
```

Comment:
Should these be swapped in order?

Reply:
It does not matter, I'll wait for all messages anyway. Swapped just for convenience.